packages/yarnpkg-core/sources/Configuration.ts

```diff
@@ -957,30 +958,45 @@ function getRcFilename() {
   return DEFAULT_RC_FILENAME as Filename;
 }
 
+async function tryRead(p: PortablePath) {
+  try {
+    return xfs.readFilePromise(p);
```

Comment:
```suggestion
    return await xfs.readFilePromise(p);
```
Otherwise you wont hit the catch